lib/reek/configuration/configuration_validator.rb

```diff
@@ -0,0 +1,35 @@
+module Reek
+  module Configuration
+    #
+    # Configuration validator module.
+    #
+    module ConfigurationValidator
+      private
+
+      def smell_type?(key)
+        Reek::Smells.const_get key
+      rescue NameError
+        false
+      end
+
+      def error_message_for_missing_directory(pathname)
+        "Configuration error: Directory `#{pathname}` does not exist"
+      end
+
+      def error_message_for_file_given(pathname)
+        "Configuration error: `#{pathname}` is supposed to be a directory but is a file"
+      end
+
+      def validate_directory(pathname)
+        abort(error_message_for_missing_directory(pathname)) unless pathname.exist?
+        abort(error_message_for_file_given(pathname)) if pathname.file?
+      end
```

Comment:
Why `abort` here (which is effectively `exit 1` + a message to stderr) rather than a regular `raise`?